src/main/java/org/folio/services/invoice/InvoiceCancelService.java

```diff
@@ -149,8 +153,16 @@ private CompletableFuture<Void> unreleaseEncumbrances(List<InvoiceLine> invoiceL
       .collect(toList());
     if (poLineIds.isEmpty())
       return completedFuture(null);
-    return orderLineService.getPoLines(queryToGetPoLinesWithRightPaymentStatusByIds(poLineIds), requestContext)
-      .thenCompose(poLines -> selectPoLinesWithOpenOrders(poLines, requestContext))
+    List<CompletableFuture<List<PoLine>>> futureList = StreamEx
+      .ofSubLists(poLineIds, MAX_IDS_FOR_GET_RQ)
+      .map(this::queryToGetPoLinesWithRightPaymentStatusByIds)
+      .map(query -> orderLineService.getPoLines(query, requestContext))
+      .collect(toList());
+
+    CompletableFuture<List<PoLine>> poLinesFuture = collectResultsOnSuccess(futureList)
+      .thenApply(col -> col.stream().flatMap(Collection::stream).collect(toList()));
```

Comment:
After this change the resulting list of polines can contain duplicates. Please make the elements unique to avoid issues

Reply:
Added distinct() .